docs/extensions/network.md

```diff
@@ -0,0 +1,81 @@
+# Gardener Network Extension
+
+Gardener is an open-source project that provides a nested user model. Basically, there are two types of services provided by Gardener to its users:
+
+- Managed: users only request a Kubernetes cluster (Clusters-as-a-Service)
+- Hosted: users utilize Gardener to provide their own managed version of Kubernetes (Cluster-Provisioner-as-a-service)
+
+For the first set of users, the choice of network plugin might not be so important, however, for the second class of users (i.e., Hosted) it is important to be able to customize networking based on their needs.
```

Comment:
I would rephrase the first sentence - for many people it is important as it is running in their cluster and might provide higher levels of services (e.g., calico provides network policies while flannel does not).

For the second scenario ("hosted") it's quite unimportant - the only thing that the operators want to do is to offer choice for end-users. They themselves are not the primary users of shoots.

Reply:
I am not sure you got my point in the text, so I rephrased it.